### ISSUE TITLE
[IFC Tunnel] IfcGeoScienceModel and IfcGeoScienceFeature

### ISSUE THREAD

**Issue by larswik** [bSI-InfraRoom/IFC-Specification#393]
Updated UML model:
![image](https://user-images.githubusercontent.com/25506404/157014110-d264bac3-542c-4a7b-b34f-f58362ddf759.png)

![image](https://user-images.githubusercontent.com/25506404/157014139-eeeb2983-c7d5-440c-8fee-3314df2fe5f9.png)

_Originally posted by @larswik in https://github.com/bSI-InfraRoom/IFC-Specification/issues/315#issuecomment-1060458394_

**Comment by larswik** [comment]
New abstract Entity IfcGeoScienceElement subtype of IfcElement : An abstract entity for geotechnical and geological concepts.
New entity IfcGeoScienceFeature subtype of IfcGeoScienceElement : Represents a geological or geotechnical feature as an interpretation of factual data such as observations, measurements and tests.
PDTs:
- DISCRETEDISCONTINUITY : Any interuption of the continuity in the rock material with its attendant mechanical, hydraulic and thermal properties.
- FOLD : A fold is formed by one or more systematically curved layers, surfaces, or lines in a rock body. A fold denotes a structure formed by the deformation of a geologic structure, such as a contact which the original undeformed geometry is presumed, to form a structure that may be described by the translation of an abstract line (the fold axis) parallel to itself along some curvilinear path (the fold profile). Folds have a hinge zone (zone of maximum curvature along the surface) and limbs (parts of the deformed surface not in the hinge zone).  Folds are described by an axial surface, hinge line, profile geometry, the solid angle between the limbs, and the relationships between adjacent folded surfaces if the folded structure is a Layering fabric. 
- FLUIDBODY : A distinct body of some fluid (liquid, gas) that fills the voids of a container such as an aquifer, system of aquifers, water well, etc. In hydrogeology this body is usually constituted by groundwater, but the model allows for other types of fillers e.g. petroleum.
- PIEZOMETRICWATERLEVEL : A surface on a fluid body within a local or regional area, e.g. piezometric, potentiometric, water table, salt wedge, etc.
- VOIDBODY : A discrete air filled geological feature, including caves and other voids
- GEOLOGICALUNIT : a GeologicUnit may represent a body of material in the Earth whose complete and precise extent is inferred to exist (e.g., North American Data Model GeologicUnit, Stratigraphic unit in the sense of NACSN, or International Stratigraphic Code ), or a classifier used to characterize parts of the Earth (e.g. lithologic map unit like 'granitic rock' or 'alluvial deposit', surficial units like 'till' or 'old alluvium'). It includes both formal units (i.e. formally adopted and named in an official lexicon) and informal units (i.e. named but not promoted to a lexicon) and unnamed units (i.e., recognizable, described and delineable in the field but not otherwise formalised).  In simpler terms, a geologic unit is a package of earth material (generally rock or soil).
- GEOTECHNICALUNIT : A surface or a volume in which the mechanical behaviour and other design-relevant characteristics are characterized using the same  geotechnical parameters values. Several alternative classifications (=GeotechModels) can be required in a project for different design tasks.
- HAZARDAREA : Discrete spatial objects representing a natural hazard.
- HYDROGEOUNIT : Any soil or rock unit or zone that by virtue of its hydraulic properties has a distinct influence on the storage or movement of groundwater (after ANS, 1980).
- FAULT : A shear displacement structure includes all brittle to ductile style structures along which displacement has occurred, from a simple, single 'planar' brittle or ductile surface to a fault system comprised of tens of strands of both brittle and ductile nature. This structure may have some significant thickness (a deformation zone) and have an associated body of deformed rock that may be considered a deformation unit (which geologicUnitType is ‘DeformationUnit’) which can be associated to the ShearDisplacementStructure using GeologicFeatureRelation from the GeoSciML Extension package
- CONTACT : A contact is a general concept representing any kind of surface separating two geologic units, including primary boundaries such as depositional contacts, all kinds of unconformities, intrusive contacts, and gradational contacts, as well as faults that separate geologic units.
- PHYSICALPROPERTYDISTRIBUTION : Additional option (alternative to discrete models) to describe a  +/- continuous spatial distribution of any physical parameter (geotechnical key-parameters, permeabilty, likelyhood of. e.g. a fault or any other uncertainty-related information)

New Entity IfcGeoScienceModel subtype of IfcGeoScienceElement : Model of geological structured as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and GeoHazardModel
PDTs:
- GEOTECHMODEL : Model with project and design-task specific  geotechnical classification (e.g. material with similar characteristics regarding geomechanical properties or excavation)
- HYDROGEOMODEL : Model of hydrogeological conditions with parameters like permeability and definition of boundary conditions for hydrogeological modelling
- GEOLOGYMODEL : Model of geological structured as considered relevant for the project, as a base for the definition of building- and design-related geotechnical models, hydrogeological models and GeoHazardModel
- GEOTECHSYNTHESISMODEL : Link between the design and modelled geology and geotechnical conditions:
summarized interpretation with regard to building, construction method,… in relation to a section of the alignment or building structure. Typical definition of "baseline conditions" as usually included in a geotech. longitudinal section
- PHYSICALPROPERTYDISTRIBUTIONMODEL : Additional option (alternative to discrete models) to describe a  +/- continuous spatial distribution of any physical properties (geotechnical key-parameters, permeabilty, likelyhood of. e.g. a fault or any other uncertainty-related information)
- GEOHAZARDMODEL : Model of natural ("geogenic") hazards like rock fall, avalanches, seismicity,...

**Comment by larswik** [comment]
To have this change independent, we should also make the changes for IfcBorehole:
- Change baseclass to IfcGeoScienceElement
- Add PredefinedTypes:
  - COREDRILLING : A drilling process in which borehole is performed by cutting out cylindrical rock or soil samples in the field, using a core barrel.
   - DESTRUCTIVEDRILLING : A drilling process in which the boring is performed using destructive tools
  - TRIALPIT : An excavation made for the purpose of observing shallow subsurface conditions, performing field tests and obtaining soil samples.

**Comment by pjanck** [comment]
Resolved through #438
